src/domain/dataset-env-vars/domain/src/repos/dataset_env_var_repository.rs

```diff
@@ -0,0 +1,123 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use internal_error::InternalError;
+use opendatafabric::DatasetID;
+use thiserror::Error;
+use uuid::Uuid;
+
+use crate::DatasetEnvVar;
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[async_trait::async_trait]
+pub trait DatasetEnvVarRepository: Send + Sync {
+    async fn save_dataset_env_var(
+        &self,
+        dataset_env_var: &DatasetEnvVar,
+    ) -> Result<(), SaveDatasetEnvVarError>;
+
+    async fn get_all_dataset_env_vars_count_by_dataset_id(
+        &self,
+        dataset_id: &DatasetID,
+    ) -> Result<usize, GetDatasetEnvVarError>;
+
+    async fn get_all_dataset_env_vars_by_dataset_id(
+        &self,
+        dataset_id: &DatasetID,
+        pagination: &DatasetEnvVarPaginationOpts,
+    ) -> Result<Vec<DatasetEnvVar>, GetDatasetEnvVarError>;
+
+    async fn get_dataset_env_var_by_id(
+        &self,
+        dataset_env_var_id: &Uuid,
+    ) -> Result<DatasetEnvVar, GetDatasetEnvVarError>;
+
+    async fn get_dataset_env_var_by_key_and_dataset_id(
+        &self,
+        dataset_env_var_key: &str,
+        dataset_id: &DatasetID,
+    ) -> Result<DatasetEnvVar, GetDatasetEnvVarError>;
+
+    async fn delete_dataset_env_var(
+        &self,
+        dataset_env_var_id: &Uuid,
+    ) -> Result<(), DeleteDatasetEnvVarError>;
+
+    async fn modify_dataset_env_var(
+        &self,
+        dataset_env_var_id: &Uuid,
+        new_value: Vec<u8>,
+        secret_nonce: Option<Vec<u8>>,
+    ) -> Result<(), ModifyDatasetEnvVarError>;
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+pub struct DatasetEnvVarPaginationOpts {
+    pub limit: i64,
+    pub offset: i64,
+}
```

Comment:
This isn't specific to env vars. Could be useful for reuse in `database-common`. Look for similar structs and replace them too - I guess we had one in access tokens topic.